modules/transcription-service-google-speech-impl/src/main/java/org/opencastproject/transcription/googlespeech/GoogleSpeechTranscriptionService.java

```diff
@@ -866,6 +870,37 @@ private String buildResultsFileName(String jobId) {
     return PathSupport.toSafeName(jobId + ".json");
   }
 
+  private void cancelTranscription(String jobId, String message) {
+    try {
+      database.updateJobControl(jobId, TranscriptionJobControl.Status.Canceled.name());
+      String mpId = database.findByJob(jobId).getMediaPackageId();
+      try {
+        // Delete file stored on Google storage
+        String token = getRefreshAccessToken();
+        deleteStorageFile(mpId, token);
+      } catch (Exception ex) {
+        logger.warn(String.format("could not delete file %s.%s from Google cloud storage", mpId, defaultEncoding), ex);
+      }
+      // Send notification email
+      sendEmail("Transcription ERROR", String.format("%s(media package %s, job id %s).", message, mpId, jobId));
+    } catch (Exception e) {
+      logger.error(String.format("ERROR while deleting transcription job: %s", jobId), e);
+    }
+  }
+
+  private boolean hasTranscriptionRequestExpired(String jobId) {
+    try {
+      // set a time limit based on video duration and maximum processing time
+      if (database.findByJob(jobId).getDateCreated().getTime() + database.findByJob(jobId).getTrackDuration()
+              + (completionCheckBuffer + maxProcessingSeconds) * 1000 < System.currentTimeMillis()) {
+        return true;
+      }
+    } catch (Exception e) {
+      logger.error(String.format("ERROR while calculating transcription request expiration for job: %s", jobId), e);
+    }
+    return false;
+  }
```

Comment:
Would it make sense to return true when there is an ERROR level error? Or is it useful to allow manual intervention to troubleshot the issue before cancelling the transcript? I'm wonder why it is better to leave the job in perpetual non-expired state during an exception. For example, maybe an issue with the database connection at that point or the transcription job has not been fully at this point.

Reply:
yes, returning `true` will avoid perpetual non-expired state.